torch/__init__.py

```diff
@@ -1328,6 +1328,7 @@ def __call__(self, model_, inputs_):
 def compile(model: Optional[Callable] = None, *,
             fullgraph: builtins.bool = False,
             dynamic: builtins.bool = False,
+            dynamic_args = None,
```

Comment:
will need dynamic_kwargs too, or maybe a struct that can be passed in here. Type annotation here extremely useful

Reply:
I think we really want per-dimension tagging, not just at the arg level.  We could reuse some FX type annotations:
```
from torch.fx.tensor_type import TensorType, Dyn

def foo(x: TensorType[Dyn, 100]):
   ...
```

Reply:
Sherlock's design does support per dim